src/features/terraformVersion.ts

```diff
@@ -0,0 +1,82 @@
+import * as vscode from 'vscode';
+import * as terraform from '../terraform';
+import { ClientCapabilities, ServerCapabilities, StaticFeature } from 'vscode-languageclient';
+import { getActiveTextEditor } from '../utils/vscode';
+import { ExperimentalClientCapabilities } from './types';
+import { Utils } from 'vscode-uri';
+import TelemetryReporter from '@vscode/extension-telemetry';
+import { LanguageClient } from 'vscode-languageclient/node';
+
+export class TerraformVersionFeature implements StaticFeature {
+  private disposables: vscode.Disposable[] = [];
+
+  private clientTerraformVersionCommandId = 'client.refreshTerraformVersion';
+
+  private terraformStatus = vscode.languages.createLanguageStatusItem('terraform.status', [
+    { language: 'terraform' },
+    { language: 'terraform-vars' },
+  ]);
+
+  constructor(
+    private client: LanguageClient,
+    private reporter: TelemetryReporter,
+    private outputChannel: vscode.OutputChannel,
+  ) {
+    this.terraformStatus.name = 'Terraform';
+    this.terraformStatus.detail = 'Terraform Version';
+    this.disposables.push(this.terraformStatus);
+  }
+
+  public fillClientCapabilities(capabilities: ClientCapabilities & ExperimentalClientCapabilities): void {
+    capabilities.experimental = capabilities.experimental || {};
+    capabilities.experimental.refreshTerraformVersionCommandId = this.clientTerraformVersionCommandId;
+  }
+
+  public async initialize(capabilities: ServerCapabilities): Promise<void> {
+    if (!capabilities.experimental?.refreshTerraformVersion) {
+      this.outputChannel.appendLine("Server doesn't support client.refreshTerraformVersion");
+      return;
+    }
+
+    await this.client.onReady();
+
+    const handler = this.client.onRequest(this.clientTerraformVersionCommandId, async () => {
+      const editor = getActiveTextEditor();
+      if (editor === undefined) {
+        return;
+      }
+
+      const moduleDir = Utils.dirname(editor.document.uri);
+
+      try {
+        const response = await terraform.terraformVersion(moduleDir.toString(), this.client, this.reporter);
+        this.terraformStatus.text = response.discovered_version;
```

Comment:
What do we want to display if no version was discovered?

Reply:
Not sure here. If I don't set it, its an empty space. I could replace with `N/A` or `Not found` but it looks odd.

You can't hide a LanguageStatusItem that's been created like you can with a StatusBarItem, so you are left with it there after initialization. The alternative is to not create it unless a version comes back, but that does not account for the scenario when one directory has a terraform binary available to query, and another doesn't.

Reply:
I think the combination of what you're proposing may work best, i.e. not create the `LanguageStatusItem` at all if the version is unknown and display something like `unknown` if it changes to unknown as part of the refresh.

FWIW unlike the constraint, we don't really re-discover the installed version on LS side at the moment, so the scenario with `unknown` basically can not happen, for now. https://github.com/hashicorp/terraform-ls/issues/69